lib/message/encrypt.js

```diff
@@ -15,5 +15,24 @@ export default function encryptMessage(options) {
     options.date = typeof options.date === 'undefined' ? serverTime() : options.date;
     options.compression = options.compression ? openpgp.enums.compression.zlib : undefined;
 
-    return openpgp.encrypt(options);
+    if (options.detached) {
+        // Create detached signature of message
+        const signature = await options.message.signDetached(options.privateKeys);
+
+        // Encrypt message without signing it (thus no need to change .detached)
+        options.privateKeys = [];
+        const result = await openpgp.encrypt(options);
```

Comment:
And then also here, you could do
```suggestion
        const result = await openpgp.encrypt({
            ...options,
            privateKeys: []
        });
```